shared/appengine.go

```diff
@@ -86,27 +86,31 @@ func (c *clientsImpl) Init(ctx context.Context) (err error) {
 // Close closes all clients in Clients. It must be called once and only once
 // before the server exits. Do not use AppEngineAPI afterwards.
 func (c *clientsImpl) Close() {
+	log.Printf("Closing clients")
```

Comment:
Nit; could be `Print` or `Println` since you don't have format arguments, I think? (Both appear equivalent for `log`, which always adds a newline)

Reply:
```suggestion
	log.Println("Closing clients")
```